### ISSUE TITLE
Make the center content take full width

### ISSUE THREAD

**Issue by insin** [insin/tweak-new-twitter#25]
Request from the 🍊🕸:

https://news.ycombinator.com/item?id=23879926

This seems to do the trick…

```js
document.querySelector('[data-testid="primaryColumn"]').style.maxWidth = 'none'
document.querySelector('[data-testid="primaryColumn"]').firstElementChild.lastElementChild.style.maxWidth = 'none'
document.querySelector('[data-testid="sidebarColumn"]').style.display = 'none'
```

…but it makes media tweets absolutely massive:

![lpOo8UnVcL](https://user-images.githubusercontent.com/226692/87866554-993f6e00-c9c6-11ea-8c9c-7e737bd7d9f7.png)

**Comment by neberej** [comment]
Hey, that was me. :) 

Those three lines of code stretches the main content to full width but it also hides search bar (because search bar is in the sidebar).

We could replace the Twitter logo with the search box.

```
document.querySelector('h1[role="heading"]').append(document.querySelector('form[aria-label="Search Twitter"]'));
```


**Final result:**
![image](https://user-images.githubusercontent.com/6032189/87867917-62c20e80-c946-11ea-9a7f-32e327f6208c.png)

**Comment by bellyrose** [comment]
Is it possible to add this to the Addon manually? Would love to get rid of the empty space that's left behind

**Comment by insin** [comment]
The search doesn't work on the left hand side (the dropdown goes under the nav menu and also causes a scrollbar to appear), what should we do with it, bearing in mind that the top bar can also contain the shared tweets timeline and the sparkle menu (when the Home timeline is enabled)

![firefox_QP5AiPIzWG](https://user-images.githubusercontent.com/226692/128649623-162f9f32-593b-4883-a245-4b42fd58da8c.png)

![BdsvFv1dWd](https://user-images.githubusercontent.com/226692/128649622-5c9d5a60-c5c3-4614-8191-602349e460d3.png)

![0PucXONUoR](https://user-images.githubusercontent.com/226692/128649621-59e2e12d-5b9a-46f1-b9f1-402f7a004192.png)

![bldIR5350O](https://user-images.githubusercontent.com/226692/128649619-53a9d1af-7d3e-4679-85af-5f8c8fd97cab.png)

**Comment by insin** [comment]
Also adding a li'l slider to let you control the width used, but it's a bit unsatisfying when view in a larger window, as the widest and 2nd widest (nav bar collapsed to icons) versions of the layout are designed around leaving a set amount of space for the sidebar.

![HD0KBPOh8T](https://user-images.githubusercontent.com/226692/128684914-561e7c32-fa7b-4a13-90a3-77c02d16deae.gif)

We could do a bit of fiddling with the header/main layout to make it centred instead of leaving that big gap for the sidebar, but there's nothing we can really do about the breakpoint it switches to the no sidebar version of the layout at.

**Comment by insin** [comment]
Released in v2.9
